ledger/store/trackerdb/testsuite/utils_test.go

```diff
@@ -0,0 +1,440 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package testsuite
+
+// A collection of utility functions and types to write the tests in this module.
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/ledger/store/trackerdb"
+	"github.com/algorand/go-algorand/ledger/store/trackerdb/generickv"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/algorand/go-algorand/util/db"
+	"github.com/stretchr/testify/require"
+)
+
+type customT struct {
+	db    dbForTests
+	proto config.ConsensusParams
+	*testing.T
+}
+
+type dbForTests interface {
+	// generickv.KvWrite
+	// generickv.KvRead
+	trackerdb.Store
+}
+
+type genericTestEntry struct {
+	name string
+	f    func(*customT)
+}
+
+// list of tests to be run on each KV DB implementation
+var genericTests []genericTestEntry
+
+// registerTest registers the given test with the suite
+func registerTest(name string, f func(*customT)) {
+	genericTests = append(genericTests, genericTestEntry{name, f})
+}
+
+// runGenericTestsWithDB runs a generic set of tests on the given database
+func runGenericTestsWithDB(t *testing.T, dbFactory func(config.ConsensusParams) (db dbForTests)) {
```

Comment:
what about wrapping this and running sql and kv, and then comparing data in both? like to check other side effects that are out of test asserts?

Reply:
The `dualdriver` does exactly that, it is also plugged in into this tests. 
We are still missing the PR that brings that in, after we have the `pebbledriver` PR.